foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiAssistantMessage.java

```diff
@@ -15,29 +19,48 @@
 /**
  * Represents a chat message as 'assistant' to OpenAI service.
  *
+ * <p>When {@link OpenAiAssistantMessage} is received from {@link OpenAiChatCompletionResponse}, it
+ * may contain tool calls that need to be executed. The tool calls are represented as {@link
+ * OpenAiToolCall}.
+ *
  * @since 1.4.0
  */
 @Beta
 @Value
 @Accessors(fluent = true)
 @AllArgsConstructor(access = PACKAGE)
-class OpenAiAssistantMessage implements OpenAiMessage {
+public class OpenAiAssistantMessage implements OpenAiMessage {
```

Comment:
```suggestion
class OpenAiAssistantMessage implements OpenAiMessage {
```
No one needs it public?

Reply:
Unfortunately, we need to expose the class now as the user should be able to do 
`final var functionCall = (OpenAiFunctionCall) assistantMessage.toolCalls();`

We could think why not have a method `getTools()` as part of `OpenAiChatCompletionResponse` instead as its already public/exposed. I just assumed it would be more intuitive for user to do

```
var assistantMessage = response.getMessage();
messages.add(assistantMessage)

functionCall = (OpenAiFunctionCall) assistantMessage.toolCalls()
```

instead of 
```
var assistantMessage = response.getMessage();
messages.add(assistantMessage)

functionCall = (OpenAiFunctionCall) response.getToolCalls()
```

The second one has additional risks of bloating `OpenAiChatCompletionResponse` class, especially if assistant message gets even more fields.